### PR TITLE
Refresh Coveo Token based on Expiration Time

### PR DESCRIPTION
**Pull request by varuna13**
Please provide the Jira Issue your PR is for.
Please also provide test paths following the template below.

Jira ID:  https://jira.corp.adobe.com/browse/EXLM-2582

> NOTE: `- After: https://exlm-2582--exlm--adobe-experience-league.aem.page` will be automatically replaced with the proper origin (using your branch) to test performance against with AEM Sync Bot.

Test URLs:

- Before: https://main--exlm--adobe-experience-league.aem.page

- After: https://exlm-2582--exlm--adobe-experience-league.aem.page?martech=off

### CODE REVIEW

File: scripts/data-service/coveo/coveo-token-service.js [hunk 1]
```diff
@@ -6,7 +6,17 @@ import { isSignedInUser } from '../../auth/profile.js';
...
+ * Decodes a Coveo token to extract its expiration time (`exp`).
+ * @param {string} token - The JSON Web Token (JWT) as a string.
+ * @returns {number} The expiration time (`exp`) as a Unix timestamp.
+ */
+
+function decodeCoveoTokenValidity(token) {
+  const tokenPayload = token.split('.')[1]; // Get the payload
+  const jsonPayload = JSON.parse(atob(tokenPayload)); // Decode the Base64-encoded token payload and parse it into a JSON object.
+  return jsonPayload.exp;
+}
```
Comment (noormd08) [inline scripts/data-service/coveo/coveo-token-service.js:19]: Add a null check for token

Comment (varuna13) [reply scripts/data-service/coveo/coveo-token-service.js:19]: PR updated

File: scripts/data-service/coveo/coveo-token-service.js [hunk 2]
```diff
@@ -94,29 +103,30 @@ async function fetchAndStoreCoveoToken() {
...
+      coveoResponseToken = '';
+      return storedCoveoToken;
+    }
+    const token = await fetchAndStoreCoveoToken();
+    coveoTokenExpirationTime = decodeCoveoTokenValidity(token);
+    return token;
+  }
+
   coveoResponseToken =
     coveoResponseToken ||
```
Comment (noormd08) [inline scripts/data-service/coveo/coveo-token-service.js:131]: As discussed, one of the negative case which may fail over here. Please check and update the relevant code.

Comment (varuna13) [reply scripts/data-service/coveo/coveo-token-service.js:127]: PR updated

**Review by noormd08 — APPROVED** [review]

**Pull request merged**